contracts/cw20-base/src/contract.rs

```diff
@@ -497,10 +497,10 @@ mod tests {
         let info = mock_info(&HumanAddr("creator".to_string()), &[]);
         let env = mock_env();
         let res = instantiate(deps.as_mut(), env.clone(), info.clone(), instantiate_msg);
-        match res.unwrap_err() {
-            StdError::GenericErr { msg, .. } => assert_eq!(&msg, "Initial supply greater than cap"),
-            e => panic!("Unexpected error: {}", e),
-        }
+        assert_eq!(
+            StdError::generic_err("Initial supply greater than cap"),
```

Comment:
Does this work with the backtraces enabled (or do we ever even enable them here)? It seems you left the match for StdError above. I'm fine with whatever works, let's just keep it unified.

Reply:
It does.

`matches!` cannot be used because of the "Initial supply greater than cap" msg, as it's not possible to create a `String` inside a pattern.
To be clear, something like
```
assert!(matches!(res.unwrap_err(), StdError::GenericErr { .. }));
```
works, whereas something like
```
assert!(matches!(res.unwrap_err(), StdError::GenericErr { msg: "Initial supply greater than cap", .. }));
```
does not compile, because the `msg` string is not a `String` but a `&str`, and converting to `String` is not allowed inside patterns.

So, we either use the `matches!` syntax here, and lose matching precision, or we stick with the current version, every time we want to match an explicit message / error content.